### PR TITLE
Make `label` prop required for VIconButton

### LINKED ISSUE #618: Add required `aria-label` prop to VIconButton
## Problem

The IconButton component should always have an aria- label unless it is `aria-hidden`, which it shouldn't probably ever be considering it's an interactive part of the page. Without the label screen readers don't have the necessary information to describe the button to the user.

## Description

We should add a required prop for the `aria-label` attribute to enforce that it is always passed when using this component.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #618 by @sarayourfriend

## Description

This PR adds a required prop named `label` to `VIconButton` to make sure that a button always has an accessible name. It uses a `label` prop instead of `aria-label` attr to make the validation easier, especially for elements that wrap around `VIconButton`.

## Testing Instructions

Try removing the `label` prop for one of the `VIconButton`s. You should see a warning in the console.
The CI failures will be fixed by #1137.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
This is a good approach to requiring a standard html attribute. LGTM.

File: frontend/src/components/VIconButton/VIconButton.vue [hunk 1]
```diff
@@ -63,6 +64,13 @@ export default defineComponent({
       type: Object as PropType<ButtonProps>,
       default: () => ({ variant: "plain" }),
     },
+    /**
+     * the label for the button; This is used for accessibility purposes.
+     */
+    label: {
+      type: String,
```
Comment (dhruvkb) [inline frontend/src/components/VIconButton/VIconButton.vue:71]: If we make the `label` prop accept translate results, we can drop the `.toString` from all the different places where it is passed.

```typescript
import { TranslateResult } from 'vue-i18n'
```

```suggestion
      type: [String, Object] as PropType<string | TranslateResult>,
```

File: frontend/src/components/VHeader/VHeaderMobile/VSearchBarButton.vue [hunk 2]
```diff
@@ -66,6 +67,13 @@ export default defineComponent({
       type: Boolean,
       default: false,
     },
+    /**
+     * The label to use as accessible name for the button (aria-label).
+     */
+    label: {
+      type: String,
```
Comment (dhruvkb) [inline frontend/src/components/VHeader/VHeaderMobile/VSearchBarButton.vue:74]: If we make the `label` prop accept translate results, we can drop the `.toString` from all the different places where it is passed.

```typescript
import { TranslateResult } from 'vue-i18n'
```

```suggestion
      type: [String, Object] as PropType<string | TranslateResult>,
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
There is a scope to simplify the changes.

**Comment by dhruvkb** [comment]
@obulat I think you're missing an `import type` in these commits^.

**Comment by obulat** [comment]
> @obulat I think you're missing an `import type` in these commits^.

Updated locally. It's so much cleaner now, thank you for this improvement, @dhruvkb !

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**